# Objectifs

- Pour l'année 2017

- Trouver la proportion de personnes en état de sous-nutrition ;

- Trouver le nombre théorique de personnes qui pourraient être nourries ; 

- Trouver la disponibilité alimentaire des produits végétaux ;

- Trouver l’utilisation de la disponibilité intérieure, l’alimentation animale, l'alimentation humaine.

Importation des outils pour réaliser les calcules demandé.

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

- Pandas est une librairie qui nous permet d'utiliser des fonction de calcules afin de nous ramener à nos objectifs ainsi que Numpy.
- Seaborn va nous permettre de visualiser les données plus clairement, grace à matplotlip qui permet également de visualiser les données sous formes de graphs et d'autres optiques visuels.

Importation des données à analyser.

In [2]:
population = pd.read_csv('population.csv')
sous_nutrition = pd.read_csv('sous_nutrition.csv')
dispo_alimentaire = pd.read_csv('dispo_alimentaire.csv')
aide_alimentaire = pd.read_csv('aide_alimentaire.csv')

Nos jeux de données contiennent :
- Le nombre de population dans chaque pays pour une période donnée.
- Le nombre de personnes en sous_nutrition par pays et par période.
- La disponibilité alimentaire pour l'année 2017 par pays, le produit, la disponibilité de la composition de ce dernier en 4 variables, 10 variables utiles pour connaitre le moyen d'aquisition de la disponibilité alimentaire et l'origine du produit (animale ou végétale).
- Les pays bénéficiaires de l'aide alimentaire, le produit et sa quantité.

## Identifier les types de mes variables et Corriger le format des colonnes pour chaques fichiers.

#### Fichier 1

In [3]:
population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Zone    1416 non-null   object 
 1   Année   1416 non-null   int64  
 2   Valeur  1416 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 33.3+ KB


- Dans notre jeu de données, on n'a pas des valeurs manquantes : il y a 1416 lignes pour chaques variables
- Les types de variables sont corrects

Je vais procéder à la renomination des noms de colonne :

In [4]:
population = population.rename(columns={"Zone": "pays", "Année": "annee", "Valeur": "nb_population"})
population.head(4)

pays  annee  nb_population
0  Afghanistan   2013      32269.589
1  Afghanistan   2014      33370.794
2  Afghanistan   2015      34413.603
3  Afghanistan   2016      35383.032

j'ai donc remplacer "Zone" par "pays", "Année" par "annee" et "Valeur" par "nb_population" pour plus de clareter.

Je vais procéder à la correction de l'affichage de la valeur qui correspond au nombre de population

In [5]:
population['nb_population'] = population['nb_population'] * 1000
population.info

<bound method DataFrame.info of              pays  annee  nb_population
0     Afghanistan   2013     32269589.0
1     Afghanistan   2014     33370794.0
2     Afghanistan   2015     34413603.0
3     Afghanistan   2016     35383032.0
4     Afghanistan   2017     36296113.0
...           ...    ...            ...
1411     Zimbabwe   2014     13586707.0
1412     Zimbabwe   2015     13814629.0
1413     Zimbabwe   2016     14030331.0
1414     Zimbabwe   2017     14236595.0
1415     Zimbabwe   2018     14438802.0

[1416 rows x 3 columns]>

Je vais procéder a rendre le nombre de population en chiffre entier.

In [6]:
population['nb_population'] = population['nb_population'].astype(int)
population.info

<bound method DataFrame.info of              pays  annee  nb_population
0     Afghanistan   2013       32269589
1     Afghanistan   2014       33370794
2     Afghanistan   2015       34413603
3     Afghanistan   2016       35383032
4     Afghanistan   2017       36296113
...           ...    ...            ...
1411     Zimbabwe   2014       13586707
1412     Zimbabwe   2015       13814629
1413     Zimbabwe   2016       14030331
1414     Zimbabwe   2017       14236595
1415     Zimbabwe   2018       14438802

[1416 rows x 3 columns]>

Il n y a plus de decimal.

#### Fichier 2

In [7]:
sous_nutrition.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Zone    1218 non-null   object
 1   Année   1218 non-null   object
 2   Valeur  624 non-null    object
dtypes: object(3)
memory usage: 28.7+ KB


- Les types de données sont incorrect.
- Il y a des valeurs manquantes dans la colonne "Valeur".

Je vais procéder à renomer les collonnes.

In [8]:
sous_nutrition = sous_nutrition.rename(columns={"Zone": "pays", "Année": "annee", "Valeur": "nb_ss_alimente"})
sous_nutrition.head(5)

pays      annee nb_ss_alimente
0  Afghanistan  2012-2014            8.6
1  Afghanistan  2013-2015            8.8
2  Afghanistan  2014-2016            8.9
3  Afghanistan  2015-2017            9.7
4  Afghanistan  2016-2018           10.5

In [9]:
sous_nutrition.dtypes

pays              object
annee             object
nb_ss_alimente    object
dtype: object

In [10]:
# Les périodes couvertes
sous_nutrition['annee'].unique()

array(['2012-2014', '2013-2015', '2014-2016', '2015-2017', '2016-2018',
       '2017-2019'], dtype=object)

In [11]:
sous_nutrition['annee'] = sous_nutrition['annee'].replace(['2012-2014', '2013-2015', '2014-2016', '2015-2017', '2016-2018', '2017-2019'],[2013, 2014, 2015, 2016, 2017, 2018])


In [12]:
# L'année' doit être un entier
sous_nutrition['annee'] = sous_nutrition['annee'].astype(int)
sous_nutrition.dtypes

pays              object
annee              int64
nb_ss_alimente    object
dtype: object

In [13]:
sous_nutrition.head(5)

pays  annee nb_ss_alimente
0  Afghanistan   2013            8.6
1  Afghanistan   2014            8.8
2  Afghanistan   2015            8.9
3  Afghanistan   2016            9.7
4  Afghanistan   2017           10.5

In [14]:
# Il existe une valeur abérante : <0.1 dans le nombre total
sous_nutrition['nb_ss_alimente'].unique()

array(['8.6', '8.8', '8.9', '9.7', '10.5', '11.1', '2.2', '2.5', '2.8',
       '3', '3.1', '3.3', '0.1', '1.3', '1.2', nan, '7.6', '6.2', '5.3',
       '5.6', '5.8', '5.7', '1.5', '1.6', '1.1', '1.7', '<0.1', '21.7',
       '22.4', '23.3', '22.3', '21.5', '20.9', '0.8', '2', '1.9', '1.8',
       '0.4', '0.5', '0.3', '0.2', '3.2', '3.4', '3.6', '3.8', '2.1',
       '2.3', '2.4', '0.6', '0.7', '0.9', '3.9', '2.7', '1.4', '4.8',
       '4.6', '4.9', '5', '4.4', '4.3', '4.2', '4.5', '26.2', '24.3',
       '21.3', '21.1', '2.9', '5.1', '5.2', '5.4', '203.8', '198.3',
       '193.1', '190.9', '190.1', '189.2', '23.6', '24', '24.1', '3.7',
       '7.3', '7.8', '8.4', '9', '9.1', '10.1', '10', '10.7', '11.5',
       '11.9', '11.8', '8.7', '10.3', '11', '1', '5.5', '6.8', '7.9',
       '5.9', '7', '9.2', '9.4', '9.6', '6.7', '7.1', '7.2', '14.7',
       '17.4', '20.2', '22.2', '22.8', '24.6', '31.1', '28.5', '25.4',
       '24.8', '26.1', '14.5', '15.4', '16.5', '15.8', '15.7', '10.8',
       '

In [15]:
# Convertir <0.1 en 0.05 (la moyenne car la valeur existe)
sous_nutrition.loc[sous_nutrition['nb_ss_alimente'] =="<0.1", 'nb_ss_alimente'] = 0.05

In [16]:
# Vérification OK
sous_nutrition['nb_ss_alimente'].unique()

array(['8.6', '8.8', '8.9', '9.7', '10.5', '11.1', '2.2', '2.5', '2.8',
       '3', '3.1', '3.3', '0.1', '1.3', '1.2', nan, '7.6', '6.2', '5.3',
       '5.6', '5.8', '5.7', '1.5', '1.6', '1.1', '1.7', 0.05, '21.7',
       '22.4', '23.3', '22.3', '21.5', '20.9', '0.8', '2', '1.9', '1.8',
       '0.4', '0.5', '0.3', '0.2', '3.2', '3.4', '3.6', '3.8', '2.1',
       '2.3', '2.4', '0.6', '0.7', '0.9', '3.9', '2.7', '1.4', '4.8',
       '4.6', '4.9', '5', '4.4', '4.3', '4.2', '4.5', '26.2', '24.3',
       '21.3', '21.1', '2.9', '5.1', '5.2', '5.4', '203.8', '198.3',
       '193.1', '190.9', '190.1', '189.2', '23.6', '24', '24.1', '3.7',
       '7.3', '7.8', '8.4', '9', '9.1', '10.1', '10', '10.7', '11.5',
       '11.9', '11.8', '8.7', '10.3', '11', '1', '5.5', '6.8', '7.9',
       '5.9', '7', '9.2', '9.4', '9.6', '6.7', '7.1', '7.2', '14.7',
       '17.4', '20.2', '22.2', '22.8', '24.6', '31.1', '28.5', '25.4',
       '24.8', '26.1', '14.5', '15.4', '16.5', '15.8', '15.7', '10.8',
       '11

In [17]:
# Convertir en float
sous_nutrition['nb_ss_alimente'] = sous_nutrition['nb_ss_alimente'].astype(float)
sous_nutrition.dtypes

pays               object
annee               int64
nb_ss_alimente    float64
dtype: object

In [18]:
# Convertir le nombre de population qui est en million
sous_nutrition['nb_ss_alimente'] = sous_nutrition['nb_ss_alimente'] * 1000000
sous_nutrition.info

<bound method DataFrame.info of              pays  annee  nb_ss_alimente
0     Afghanistan   2013       8600000.0
1     Afghanistan   2014       8800000.0
2     Afghanistan   2015       8900000.0
3     Afghanistan   2016       9700000.0
4     Afghanistan   2017      10500000.0
...           ...    ...             ...
1213     Zimbabwe   2014             NaN
1214     Zimbabwe   2015             NaN
1215     Zimbabwe   2016             NaN
1216     Zimbabwe   2017             NaN
1217     Zimbabwe   2018             NaN

[1218 rows x 3 columns]>

In [19]:
# La nombre doit être un entier
sous_nutrition['nb_ss_alimente'] = pd.to_numeric(sous_nutrition['nb_ss_alimente'], errors='coerce').fillna(0).astype(int)

In [20]:
sous_nutrition.dtypes

pays              object
annee              int64
nb_ss_alimente     int64
dtype: object

In [21]:
sous_nutrition.info

<bound method DataFrame.info of              pays  annee  nb_ss_alimente
0     Afghanistan   2013         8600000
1     Afghanistan   2014         8800000
2     Afghanistan   2015         8900000
3     Afghanistan   2016         9700000
4     Afghanistan   2017        10500000
...           ...    ...             ...
1213     Zimbabwe   2014               0
1214     Zimbabwe   2015               0
1215     Zimbabwe   2016               0
1216     Zimbabwe   2017               0
1217     Zimbabwe   2018               0

[1218 rows x 3 columns]>

#### Fichier 3

In [22]:
dispo_alimentaire.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Zone                                                           15605 non-null  object 
 1   Produit                                                        15605 non-null  object 
 2   Origine                                                        15605 non-null  object 
 3   Aliments pour animaux                                          2720 non-null   float64
 4   Autres Utilisations                                            5496 non-null   float64
 5   Disponibilité alimentaire (Kcal/personne/jour)                 14241 non-null  float64
 6   Disponibilité alimentaire en quantité (kg/personne/an)         14015 non-null  float64
 7   Disponibilité de matière grasse en quantité (g/personne/jo

- Beaucoup de valeurs manquantes dans de nombreuses colonnes
- Les types de variables ne sont pas cohérents

inspection

In [23]:
dispo_alimentaire = dispo_alimentaire.rename(columns={"Zone": "pays", "Produit": "produit", "Origine": "origine_produit", "Aliments pour animaux": "aliments_animaux", "Autres Utilisations":"autres_utilisations", "Disponibilité alimentaire (Kcal/personne/jour)":"dispo_alim_kcal_pers_j", "Disponibilité alimentaire en quantité (kg/personne/an)":"dispo_alim_kg_pers_an", "Disponibilité de matière grasse en quantité (g/personne/jour)":"dispo_mat_grasse_g_pers_j", "Disponibilité de protéines en quantité (g/personne/jour)": "dispo_proteine_g_pers_j", "Disponibilité intérieure": "dispo_interieure", "Exportations - Quantité": "qt_export", "Importations - Quantité": "qt_import", "Nourriture": "nb_nourriture", "Pertes": "perte", "Production": "production", "Semences" : "semences", "Traitement": "traitement", "Variation de stock": "variation_stock"})
dispo_alimentaire.head(5)

pays                produit origine_produit  aliments_animaux  \
0  Afghanistan       Abats Comestible         animale               NaN   
1  Afghanistan        Agrumes, Autres        vegetale               NaN   
2  Afghanistan  Aliments pour enfants        vegetale               NaN   
3  Afghanistan                 Ananas        vegetale               NaN   
4  Afghanistan                Bananes        vegetale               NaN   

   autres_utilisations  dispo_alim_kcal_pers_j  dispo_alim_kg_pers_an  \
0                  NaN                     5.0                   1.72   
1                  NaN                     1.0                   1.29   
2                  NaN                     1.0                   0.06   
3                  NaN                     0.0                   0.00   
4                  NaN                     4.0                   2.70   

   dispo_mat_grasse_g_pers_j  dispo_proteine_g_pers_j  dispo_interieure  \
0                       0.20                     0.77              53.0   
1                       0.01                     0.02              41.0   
2                       0.01                     0.03               2.0   
3                        NaN                      NaN               0.0   
4                       0.02                     0.05              82.0   

   qt_export  qt_import  nb_nourriture  perte  production  semences  \
0        NaN        NaN           53.0    NaN        53.0       NaN   
1        2.0       40.0           39.0    2.0         3.0       NaN   
2        NaN        2.0            2.0    NaN         NaN       NaN   
3        NaN        0.0            0.0    NaN         NaN       NaN   
4        NaN       82.0           82.0    NaN         NaN       NaN   

   traitement  variation_stock  
0         NaN              NaN  
1         NaN              NaN  
2         NaN              NaN  
3         NaN              NaN  
4         NaN              NaN

- pays : chaîne de caractères
- produit : chaîne de caractères
- origine_produit : chaîne de caractères
- aliments_animaux : float
- autres_utilisations : float
- dispo_alim_kcal_pers_j : float
- dispo_alim_kg_pers_an : float
- dispo_alim_kg_pers_an : float
- dispo_mat_grasse_g_pers_j : float
- dispo_proteine_g_pers_j : float
- dispo_interieure : float
- qt_export : float
- qt_import : float
- nb_nourriture : float
- perte : float
- production : float
- semences : float
- traitement : float
- variation_stock : float

In [24]:
dispo_alimentaire.isna().mean()

pays                         0.000000
produit                      0.000000
origine_produit              0.000000
aliments_animaux             0.825697
autres_utilisations          0.647805
dispo_alim_kcal_pers_j       0.087408
dispo_alim_kg_pers_an        0.101890
dispo_mat_grasse_g_pers_j    0.244217
dispo_proteine_g_pers_j      0.259148
dispo_interieure             0.014290
qt_export                    0.216533
qt_import                    0.048254
nb_nourriture                0.101890
perte                        0.725857
production                   0.411727
semences                     0.866004
traitement                   0.853124
variation_stock              0.565780
dtype: float64

majorité des valeurs manquantes significatif.

In [25]:
dispo_alimentaire.head(15)

pays                produit origine_produit  aliments_animaux  \
0   Afghanistan       Abats Comestible         animale               NaN   
1   Afghanistan        Agrumes, Autres        vegetale               NaN   
2   Afghanistan  Aliments pour enfants        vegetale               NaN   
3   Afghanistan                 Ananas        vegetale               NaN   
4   Afghanistan                Bananes        vegetale               NaN   
5   Afghanistan           Beurre, Ghee         animale               NaN   
6   Afghanistan                  Bière        vegetale               NaN   
7   Afghanistan                    Blé        vegetale               NaN   
8   Afghanistan   Boissons Alcooliques        vegetale               NaN   
9   Afghanistan                   Café        vegetale               NaN   
10  Afghanistan     Coco (Incl Coprah)        vegetale               NaN   
11  Afghanistan                  Crème         animale               NaN   
12  Afghanistan       Céréales, Autres        vegetale               NaN   
13  Afghanistan                 Dattes        vegetale               NaN   
14  Afghanistan     Edulcorants Autres        vegetale               NaN   

    autres_utilisations  dispo_alim_kcal_pers_j  dispo_alim_kg_pers_an  \
0                   NaN                     5.0                   1.72   
1                   NaN                     1.0                   1.29   
2                   NaN                     1.0                   0.06   
3                   NaN                     0.0                   0.00   
4                   NaN                     4.0                   2.70   
5                   NaN                    23.0                   1.17   
6                   NaN                     0.0                   0.09   
7                   NaN                  1369.0                 160.23   
8                   NaN                     0.0                   0.00   
9                   NaN                     0.0                   0.00   
10                  NaN                     0.0                   0.00   
11                  NaN                     3.0                   0.60   
12                  NaN                     0.0                   0.00   
13                  NaN                     0.0                   0.02   
14                  NaN                     2.0                   0.53   

    dispo_mat_grasse_g_pers_j  dispo_proteine_g_pers_j  dispo_interieure  \
0                        0.20                     0.77              53.0   
1                        0.01                     0.02              41.0   
2                        0.01                     0.03               2.0   
3                         NaN                      NaN               0.0   
4                        0.02                     0.05              82.0   
5                        2.61                     0.03              36.0   
6                         NaN                     0.00               3.0   
7                        4.69                    36.91            5992.0   
8                         NaN                      NaN               0.0   
9                         NaN                     0.00               0.0   
10                       0.00                     0.00               0.0   
11                       0.32                     0.04              18.0   
12                       0.00                     0.00               0.0   
13                       0.00                     0.00               1.0   
14                        NaN                      NaN              16.0   

    qt_export  qt_import  nb_nourriture  perte  production  semences  \
0         NaN        NaN           53.0    NaN        53.0       NaN   
1         2.0       40.0           39.0    2.0         3.0       NaN   
2         NaN        2.0            2.0    NaN         NaN       NaN   
3         NaN        0.0            0.0    NaN         NaN       NaN   
4         NaN       82.0        

In [26]:
dispo_alimentaire['aliments_animaux'].fillna(0, inplace=True)
dispo_alimentaire['autres_utilisations'].fillna(0, inplace=True)
dispo_alimentaire['dispo_alim_kcal_pers_j'].fillna(0, inplace=True)
dispo_alimentaire['dispo_alim_kg_pers_an'].fillna(0, inplace=True)
dispo_alimentaire['dispo_mat_grasse_g_pers_j'].fillna(0, inplace=True)
dispo_alimentaire['dispo_proteine_g_pers_j'].fillna(0, inplace=True)
dispo_alimentaire['dispo_interieure'].fillna(0, inplace=True)
dispo_alimentaire['qt_export'].fillna(0, inplace=True)
dispo_alimentaire['qt_import'].fillna(0, inplace=True)
dispo_alimentaire['nb_nourriture'].fillna(0, inplace=True)
dispo_alimentaire['perte'].fillna(0, inplace=True)
dispo_alimentaire['production'].fillna(0, inplace=True)
dispo_alimentaire['semences'].fillna(0, inplace=True)
dispo_alimentaire['traitement'].fillna(0, inplace=True)
dispo_alimentaire['variation_stock'].fillna(0, inplace=True)

j'ai donc remplacer tout les Nan par zero pour avoir aucun donnée manquante et pour poursuivre mon analyze.

In [27]:
dispo_alimentaire.isna().mean()

pays                         0.0
produit                      0.0
origine_produit              0.0
aliments_animaux             0.0
autres_utilisations          0.0
dispo_alim_kcal_pers_j       0.0
dispo_alim_kg_pers_an        0.0
dispo_mat_grasse_g_pers_j    0.0
dispo_proteine_g_pers_j      0.0
dispo_interieure             0.0
qt_export                    0.0
qt_import                    0.0
nb_nourriture                0.0
perte                        0.0
production                   0.0
semences                     0.0
traitement                   0.0
variation_stock              0.0
dtype: float64

Maintenant je vais pouvoir faire mais calcule sans complication

In [28]:
dispo_alimentaire.head(5)

pays                produit origine_produit  aliments_animaux  \
0  Afghanistan       Abats Comestible         animale               0.0   
1  Afghanistan        Agrumes, Autres        vegetale               0.0   
2  Afghanistan  Aliments pour enfants        vegetale               0.0   
3  Afghanistan                 Ananas        vegetale               0.0   
4  Afghanistan                Bananes        vegetale               0.0   

   autres_utilisations  dispo_alim_kcal_pers_j  dispo_alim_kg_pers_an  \
0                  0.0                     5.0                   1.72   
1                  0.0                     1.0                   1.29   
2                  0.0                     1.0                   0.06   
3                  0.0                     0.0                   0.00   
4                  0.0                     4.0                   2.70   

   dispo_mat_grasse_g_pers_j  dispo_proteine_g_pers_j  dispo_interieure  \
0                       0.20                     0.77              53.0   
1                       0.01                     0.02              41.0   
2                       0.01                     0.03               2.0   
3                       0.00                     0.00               0.0   
4                       0.02                     0.05              82.0   

   qt_export  qt_import  nb_nourriture  perte  production  semences  \
0        0.0        0.0           53.0    0.0        53.0       0.0   
1        2.0       40.0           39.0    2.0         3.0       0.0   
2        0.0        2.0            2.0    0.0         0.0       0.0   
3        0.0        0.0            0.0    0.0         0.0       0.0   
4        0.0       82.0           82.0    0.0         0.0       0.0   

   traitement  variation_stock  
0         0.0              0.0  
1         0.0              0.0  
2         0.0              0.0  
3         0.0              0.0  
4         0.0              0.0

#### La disponibilité alimentaire est exprimé en millier de tonne, à convertit en kg

In [29]:
dispo_alimentaire['dispo_alim_kg_pers_an'] = dispo_alimentaire['dispo_alim_kg_pers_an'] * 1000000
dispo_alimentaire['dispo_alim_kg_pers_an']

0        1720000.0
1        1290000.0
2          60000.0
3              0.0
4        2700000.0
           ...    
15600    4700000.0
15601    3340000.0
15602      60000.0
15603      70000.0
15604     480000.0
Name: dispo_alim_kg_pers_an, Length: 15605, dtype: float64

In [30]:
dispo_alimentaire['nb_nourriture'] = dispo_alimentaire['nb_nourriture'] * 1000000
dispo_alimentaire['nb_nourriture']

0        53000000.0
1        39000000.0
2         2000000.0
3               0.0
4        82000000.0
            ...    
15600     3000000.0
15601     2000000.0
15602           0.0
15603           0.0
15604           0.0
Name: nb_nourriture, Length: 15605, dtype: float64

In [31]:
dispo_alimentaire['dispo_interieure'] = dispo_alimentaire['dispo_interieure'] * 1000000
dispo_alimentaire['dispo_interieure']

0        53000000.0
1        41000000.0
2         2000000.0
3               0.0
4        82000000.0
            ...    
15600     3000000.0
15601     2000000.0
15602           0.0
15603           0.0
15604           0.0
Name: dispo_interieure, Length: 15605, dtype: float64

In [32]:
dispo_alimentaire['aliments_animaux'] = dispo_alimentaire['aliments_animaux'] * 1000000
dispo_alimentaire['aliments_animaux'].describe()

count    1.560500e+04
mean     8.357866e+07
std      1.779289e+09
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      1.500000e+11
Name: aliments_animaux, dtype: float64

In [33]:
dispo_alimentaire['perte'] = dispo_alimentaire['perte'] * 1000000
dispo_alimentaire['perte'].describe()

count    1.560500e+04
mean     2.907389e+07
std      5.846722e+08
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      5.504700e+10
Name: perte, dtype: float64

In [34]:
dispo_alimentaire.head(5)

pays                produit origine_produit  aliments_animaux  \
0  Afghanistan       Abats Comestible         animale               0.0   
1  Afghanistan        Agrumes, Autres        vegetale               0.0   
2  Afghanistan  Aliments pour enfants        vegetale               0.0   
3  Afghanistan                 Ananas        vegetale               0.0   
4  Afghanistan                Bananes        vegetale               0.0   

   autres_utilisations  dispo_alim_kcal_pers_j  dispo_alim_kg_pers_an  \
0                  0.0                     5.0              1720000.0   
1                  0.0                     1.0              1290000.0   
2                  0.0                     1.0                60000.0   
3                  0.0                     0.0                    0.0   
4                  0.0                     4.0              2700000.0   

   dispo_mat_grasse_g_pers_j  dispo_proteine_g_pers_j  dispo_interieure  \
0                       0.20                     0.77        53000000.0   
1                       0.01                     0.02        41000000.0   
2                       0.01                     0.03         2000000.0   
3                       0.00                     0.00               0.0   
4                       0.02                     0.05        82000000.0   

   qt_export  qt_import  nb_nourriture      perte  production  semences  \
0        0.0        0.0     53000000.0        0.0        53.0       0.0   
1        2.0       40.0     39000000.0  2000000.0         3.0       0.0   
2        0.0        2.0      2000000.0        0.0         0.0       0.0   
3        0.0        0.0            0.0        0.0         0.0       0.0   
4        0.0       82.0     82000000.0        0.0         0.0       0.0   

   traitement  variation_stock  
0         0.0              0.0  
1         0.0              0.0  
2         0.0              0.0  
3         0.0              0.0  
4         0.0              0.0

In [35]:
dispo_alimentaire.describe()

aliments_animaux  autres_utilisations  dispo_alim_kcal_pers_j  \
count      1.560500e+04         15605.000000            15605.000000   
mean       8.357866e+07            55.432426               31.748927   
std        1.779289e+09          3013.629812              102.961194   
min        0.000000e+00             0.000000              -21.000000   
25%        0.000000e+00             0.000000                0.000000   
50%        0.000000e+00             0.000000                3.000000   
75%        0.000000e+00             0.000000               18.000000   
max        1.500000e+11        347309.000000             1711.000000   

       dispo_alim_kg_pers_an  dispo_mat_grasse_g_pers_j  \
count           1.560500e+04               15605.000000   
mean            7.830948e+06                   0.969754   
std             2.347891e+07                   3.246695   
min            -1.930000e+06                  -0.030000   
25%             1.000000e+04                   0.000000   
50%             5.200000e+05                   0.020000   
75%             4.270000e+06                   0.310000   
max             4.307600e+08                  60.760000   

       dispo_proteine_g_pers_j  dispo_interieure     qt_export     qt_import  \
count             15605.000000      1.560500e+04  15605.000000  15605.000000   
mean                  0.906513      6.311435e+08     86.649023     83.053701   
std                   3.143516      9.002564e+09    933.435208    700.099451   
min                  -0.370000     -3.430000e+09    -41.000000   -201.000000   
25%                   0.000000      0.000000e+00      0.000000      0.000000   
50%                   0.020000      6.000000e+06      0.000000      1.000000   
75%                   0.350000      7.300000e+07      4.000000     16.000000   
max                  54.970000      7.392670e+11  42797.000000  63381.000000   

       nb_nourriture         perte     production      semences  \
count   1.560500e+04  1.560500e+04   15605.000000  15605.000000   
mean    3.124805e+08  2.907389e+07     641.440564      9.912272   
std     4.242855e+09  5.846722e+08    9270.864198    194.897843   
min    -2.460000e+08  0.000000e+00       0.000000      0.000000   
25%     0.000000e+00  0.000000e+00       0.000000      0.000000   
50%     3.000000e+06  0.000000e+00       0.000000      0.000000   
75%     4.000000e+07  0.000000e+00      42.000000      0.000000   
max     4.268500e+11  5.504700e+10  739267.000000  17060.000000   

          traitement  variation_stock  
count   15605.000000     15605.000000  
mean      141.280807        -6.690292  
std      3992.567621       362.380787  
min       -19.000000    -39863.000000  
25%         0.000000         0.000000  
50%         0.000000         0.000000  
75%         0.000000         0.000000  
max    326711.000000      5284.000000

#### Fichier 4

In [36]:
aide_alimentaire.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1475 entries, 0 to 1474
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Pays bénéficiaire  1475 non-null   object
 1   Année              1475 non-null   int64 
 2   Produit            1475 non-null   object
 3   Valeur             1475 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 46.2+ KB


- dans notre jeu de données, il n'a pas des valeurs manquantes : il y a 1475 lignes pour chaques variables
- les types de variables sont corrects

In [37]:
# Renommer les colonnes
aide_alimentaire = aide_alimentaire.rename(columns={"Pays bénéficiaire": "pays", "Année": "annee", "Produit": "produit", "Valeur": "qt_aide_kg"})
aide_alimentaire.head(5)

pays  annee              produit  qt_aide_kg
0  Afghanistan   2013  Autres non-céréales         682
1  Afghanistan   2014  Autres non-céréales         335
2  Afghanistan   2013         Blé et Farin       39224
3  Afghanistan   2014         Blé et Farin       15160
4  Afghanistan   2013             Céréales       40504

In [38]:
aide_alimentaire.describe()

annee     qt_aide_kg
count  1475.000000    1475.000000
mean   2014.054237    7481.966780
std       0.946916   23339.531424
min    2013.000000       0.000000
25%    2013.000000     287.500000
50%    2014.000000    1178.000000
75%    2015.000000    4334.500000
max    2016.000000  265013.000000

In [39]:
# Convertir la quantité tonne en kg
aide_alimentaire['qt_aide_kg'] = aide_alimentaire['qt_aide_kg'] * 1000
aide_alimentaire.describe()

annee    qt_aide_kg
count  1475.000000  1.475000e+03
mean   2014.054237  7.481967e+06
std       0.946916  2.333953e+07
min    2013.000000  0.000000e+00
25%    2013.000000  2.875000e+05
50%    2014.000000  1.178000e+06
75%    2015.000000  4.334500e+06
max    2016.000000  2.650130e+08

In [40]:
aide_alimentaire['pays'].unique()

array(['Afghanistan', 'Algérie', 'Angola', 'Bangladesh', 'Bénin',
       'Bhoutan', 'Bolivie (État plurinational de)', 'Burkina Faso',
       'Burundi', 'Cambodge', 'Cameroun', 'Chine, continentale',
       'Colombie', 'Comores', 'Congo', "Côte d'Ivoire", 'Cuba',
       'Djibouti', 'Égypte', 'El Salvador', 'Équateur', 'Éthiopie',
       'Gambie', 'Géorgie', 'Ghana', 'Guatemala', 'Guinée',
       'Guinée-Bissau', 'Haïti', 'Honduras',
       "Iran (République islamique d')", 'Iraq', 'Jordanie', 'Kenya',
       'Kirghizistan', 'Lesotho', 'Liban', 'Libéria', 'Libye',
       'Madagascar', 'Malawi', 'Mali', 'Mauritanie', 'Mozambique',
       'Myanmar', 'Népal', 'Nicaragua', 'Niger', 'Ouganda', 'Pakistan',
       'Palestine', 'Philippines', 'République arabe syrienne',
       'République centrafricaine', 'République démocratique du Congo',
       'République démocratique populaire lao', 'République dominicaine',
       'République populaire démocratique de Corée',
       'République-Unie de T

In [41]:
aide_alimentaire['qt_aide_kg'].unique()

array([  682000,   335000, 39224000, ...,    96000,  5022000,  2310000])

In [42]:
aide_alimentaire['annee'].unique()

array([2013, 2014, 2015, 2016])

# Calcule des objectives demandé

### 1) La proportion de personnes en état de sous-nutrition en 2017

In [43]:
# On sélectionne la population de 2017
population_2017 = population.loc[population['annee']==2017]
total_population_2017 = population_2017['nb_population'].sum()
total_population_2017

7548134097

In [44]:
# On sélectionne la population en sous nutrition de 2017
pop_sous_nutrition_2017 = sous_nutrition.loc[sous_nutrition['annee']==2017]
total_pop_sous_nutrition_2017 = pop_sous_nutrition_2017['nb_ss_alimente'].sum()
total_pop_sous_nutrition_2017

536700000

In [45]:
# Calcul de La proportion
proportion_personnes_sous_nutrition_2017 = (total_pop_sous_nutrition_2017 / total_population_2017)*100
proportion_personnes_sous_nutrition_2017

7.110366523738774

In [46]:
# Affichage du résultat
print(f'En 2017, {round(proportion_personnes_sous_nutrition_2017,2)} % de la population mondiale était en mal nutrition')

En 2017, 7.11 % de la population mondiale était en mal nutrition


### 2) Le nombre théorique de personnes qui pourraient être nourries à partir de la disponibilité alimentaire mondiale

In [47]:
#On croise les données de la population mondiale avec la disponibilité alimentaire par pays
dispo_alim_pop_2017 = pd.merge(dispo_alimentaire, population_2017, how='left', on='pays')
dispo_alim_pop_2017

pays                produit origine_produit  aliments_animaux  \
0       Afghanistan       Abats Comestible         animale               0.0   
1       Afghanistan        Agrumes, Autres        vegetale               0.0   
2       Afghanistan  Aliments pour enfants        vegetale               0.0   
3       Afghanistan                 Ananas        vegetale               0.0   
4       Afghanistan                Bananes        vegetale               0.0   
...             ...                    ...             ...               ...   
15600  Îles Salomon       Viande de Suides         animale               0.0   
15601  Îles Salomon    Viande de Volailles         animale               0.0   
15602  Îles Salomon          Viande, Autre         animale               0.0   
15603  Îles Salomon                    Vin        vegetale               0.0   
15604  Îles Salomon         Épices, Autres        vegetale               0.0   

       autres_utilisations  dispo_alim_kcal_pers_j  dispo_alim_kg_pers_an  \
0                      0.0                     5.0              1720000.0   
1                      0.0                     1.0              1290000.0   
2                      0.0                     1.0                60000.0   
3                      0.0                     0.0                    0.0   
4                      0.0                     4.0              2700000.0   
...                    ...                     ...                    ...   
15600                  0.0                    45.0              4700000.0   
15601                  0.0                    11.0              3340000.0   
15602                  0.0                     0.0                60000.0   
15603                  0.0                     0.0                70000.0   
15604                  0.0                     4.0               480000.0   

       dispo_mat_grasse_g_pers_j  dispo_proteine_g_pers_j  dispo_interieure  \
0                           0.20                     0.77        53000000.0   
1                           0.01                     0.02        41000000.0   
2                           0.01                     0.03         2000000.0   
3                           0.00                     0.00               0.0   
4                           0.02                     0.05        82000000.0   
...                          ...                      ...               ...   
15600                       4.28                     1.41         3000000.0   
15601                       0.69                     1.14         2000000.0   
15602                       0.00                     0.04               0.0   
15603                       0.00                     0.00               0.0   
15604                       0.21                     0.15               0.0   

       qt_export  qt_import  nb_nourriture      perte  production  semences  \
0            0.0        0.0     53000000.0        0.0        53.0       0.0   
1            2.0       40.0     39000000.0  2000000.0         3.0       0.0   
2            0.0        2.0      2000000.0        0.0         0.0       0.0   
3            0.0        0.0            0.0        0.0         0.0       0.0   
4            0.0       82.0     82000000.0        0.0         0.0       0.0   
...          ...        ...            ...        ...         ...       ...   
15600        0.0        0.0      3000000.0        0.0         2.0       0.0   
15601        0.0        2.0      2000000.0        0.0         0.0       0.0   
15602        0.0        0.0            0.0        0.0         0.0       0.0   
15603        0.0        0.0            0.0        0.0         0.0       0.0   
15604        0.0        0.0            0.0        0.0         0.0       0.0   

       traitement  variation_stock   annee  nb_population  
0             0.0              0.0  2017.0     36296113.0  
1             0.0              0.0  2017.0     36296113.0  
2             0.0              0.0  2017.0     3629

In [53]:
# La disposition alimentaire en millier de tonne converti en kg
dispo_alimentaire_2017 = dispo_alimentaire['dispo_interieure'].sum()
dispo_alimentaire_2017

9848994000000.0

In [54]:
total_nb_nourriture = dispo_alimentaire['nb_nourriture'].sum()
total_nb_nourriture


4876258000000.0

In [55]:
# La disposition alimentaire en millier de tonne converti en kg
total_dispo_alim_kg_pers_an = dispo_alimentaire['dispo_alim_kg_pers_an'].sum()
total_dispo_alim_kg_pers_an

122201940000.0

In [56]:
dispo_alimentaire_2017/total_dispo_alim_kg_pers_an

80.5960527304231

In [57]:
total_dispo_alim_kcal_pers_j = dispo_alimentaire['dispo_alim_kcal_pers_j'].sum()
total_dispo_alim_kcal_pers_j

495442.0

In [58]:
# Besoin calorique par personne H/F en moyenne par jour
besoin_kcal_perosnne_moyenne_j = 2500

In [59]:
# Le nombre de personnes thérique à nourir en 2017 en kg par personne
nb_theorique_personne_a_nourir = total_dispo_alim_kcal_pers_j / besoin_kcal_perosnne_moyenne_j 
nb_theorique_personne_a_nourir

198.1768